<!DOCTYPE html>
<html lang='en'>
<head>
<title>musl-fixes.patch\chromium\community - aports - Unnamed repository; edit this file 'description' to name the repository.
</title>
<meta name='generator' content='cgit v1.2.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='//alpinelinux.org/alpine-logo.ico'/>
<link rel='alternate' title='Atom feed' href='http://git.alpinelinux.org/aports/atom/community/chromium/musl-fixes.patch?h=master' type='application/atom+xml'/>
<link rel='vcs-git' href='git://git.alpinelinux.org/aports' title='aports Git repository'/>
<link rel='vcs-git' href='https://git.alpinelinux.org/aports' title='aports Git repository'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/'><img src='//wiki.alpinelinux.org/images/alogo.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/'>index</a> : <a title='aports' href='/aports/'>aports</a></td><td class='form'><form method='get'>
<select name='h' onchange='this.form.submit();'>
<option value='1.10-stable'>1.10-stable</option>
<option value='1.9'>1.9</option>
<option value='2.0-stable'>2.0-stable</option>
<option value='2.1-stable'>2.1-stable</option>
<option value='2.2-stable'>2.2-stable</option>
<option value='2.3-stable'>2.3-stable</option>
<option value='2.4-stable'>2.4-stable</option>
<option value='2.5-stable'>2.5-stable</option>
<option value='2.6-stable'>2.6-stable</option>
<option value='2.7-stable'>2.7-stable</option>
<option value='3.0-stable'>3.0-stable</option>
<option value='3.1-stable'>3.1-stable</option>
<option value='3.10-stable'>3.10-stable</option>
<option value='3.11-stable'>3.11-stable</option>
<option value='3.12-stable'>3.12-stable</option>
<option value='3.2-stable'>3.2-stable</option>
<option value='3.3-stable'>3.3-stable</option>
<option value='3.4-stable'>3.4-stable</option>
<option value='3.5-stable'>3.5-stable</option>
<option value='3.6-stable'>3.6-stable</option>
<option value='3.7-stable'>3.7-stable</option>
<option value='3.8-stable'>3.8-stable</option>
<option value='3.9-stable'>3.9-stable</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' value='switch'/></form></td></tr>
<tr><td class='sub'>Unnamed repository; edit this file 'description' to name the repository.
</td><td class='sub right'>uwsgi</td></tr></table>
<table class='tabs'><tr><td>
<a href='/aports/about/'>about</a><a href='/aports/'>summary</a><a href='/aports/refs/'>refs</a><a href='/aports/log/community/chromium/musl-fixes.patch'>log</a><a class='active' href='/aports/tree/community/chromium/musl-fixes.patch'>tree</a><a href='/aports/commit/community/chromium/musl-fixes.patch'>commit</a><a href='/aports/diff/community/chromium/musl-fixes.patch'>diff</a><a href='/aports/stats/community/chromium/musl-fixes.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/aports/log/community/chromium/musl-fixes.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='search' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/aports/tree/'>root</a>/<a href='/aports/tree/community'>community</a>/<a href='/aports/tree/community/chromium'>chromium</a>/<a href='/aports/tree/community/chromium/musl-fixes.patch'>musl-fixes.patch</a></div><div class='content'>blob: 161e24c55a7bb680719fd878e746e8af0d80af7b (<a href='/aports/plain/community/chromium/musl-fixes.patch'>plain</a>) (<a href='/aports/blame/community/chromium/musl-fixes.patch'>blame</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
<a id='n106' href='#n106'>106</a>
<a id='n107' href='#n107'>107</a>
<a id='n108' href='#n108'>108</a>
<a id='n109' href='#n109'>109</a>
<a id='n110' href='#n110'>110</a>
<a id='n111' href='#n111'>111</a>
<a id='n112' href='#n112'>112</a>
<a id='n113' href='#n113'>113</a>
<a id='n114' href='#n114'>114</a>
<a id='n115' href='#n115'>115</a>
<a id='n116' href='#n116'>116</a>
<a id='n117' href='#n117'>117</a>
<a id='n118' href='#n118'>118</a>
<a id='n119' href='#n119'>119</a>
<a id='n120' href='#n120'>120</a>
<a id='n121' href='#n121'>121</a>
<a id='n122' href='#n122'>122</a>
<a id='n123' href='#n123'>123</a>
<a id='n124' href='#n124'>124</a>
<a id='n125' href='#n125'>125</a>
<a id='n126' href='#n126'>126</a>
<a id='n127' href='#n127'>127</a>
<a id='n128' href='#n128'>128</a>
<a id='n129' href='#n129'>129</a>
<a id='n130' href='#n130'>130</a>
<a id='n131' href='#n131'>131</a>
<a id='n132' href='#n132'>132</a>
<a id='n133' href='#n133'>133</a>
<a id='n134' href='#n134'>134</a>
<a id='n135' href='#n135'>135</a>
<a id='n136' href='#n136'>136</a>
<a id='n137' href='#n137'>137</a>
<a id='n138' href='#n138'>138</a>
<a id='n139' href='#n139'>139</a>
<a id='n140' href='#n140'>140</a>
<a id='n141' href='#n141'>141</a>
<a id='n142' href='#n142'>142</a>
<a id='n143' href='#n143'>143</a>
<a id='n144' href='#n144'>144</a>
<a id='n145' href='#n145'>145</a>
<a id='n146' href='#n146'>146</a>
<a id='n147' href='#n147'>147</a>
<a id='n148' href='#n148'>148</a>
<a id='n149' href='#n149'>149</a>
<a id='n150' href='#n150'>150</a>
<a id='n151' href='#n151'>151</a>
<a id='n152' href='#n152'>152</a>
<a id='n153' href='#n153'>153</a>
<a id='n154' href='#n154'>154</a>
<a id='n155' href='#n155'>155</a>
<a id='n156' href='#n156'>156</a>
<a id='n157' href='#n157'>157</a>
<a id='n158' href='#n158'>158</a>
<a id='n159' href='#n159'>159</a>
<a id='n160' href='#n160'>160</a>
<a id='n161' href='#n161'>161</a>
<a id='n162' href='#n162'>162</a>
<a id='n163' href='#n163'>163</a>
<a id='n164' href='#n164'>164</a>
<a id='n165' href='#n165'>165</a>
<a id='n166' href='#n166'>166</a>
<a id='n167' href='#n167'>167</a>
<a id='n168' href='#n168'>168</a>
<a id='n169' href='#n169'>169</a>
<a id='n170' href='#n170'>170</a>
<a id='n171' href='#n171'>171</a>
<a id='n172' href='#n172'>172</a>
<a id='n173' href='#n173'>173</a>
<a id='n174' href='#n174'>174</a>
<a id='n175' href='#n175'>175</a>
<a id='n176' href='#n176'>176</a>
<a id='n177' href='#n177'>177</a>
<a id='n178' href='#n178'>178</a>
<a id='n179' href='#n179'>179</a>
<a id='n180' href='#n180'>180</a>
<a id='n181' href='#n181'>181</a>
<a id='n182' href='#n182'>182</a>
<a id='n183' href='#n183'>183</a>
<a id='n184' href='#n184'>184</a>
<a id='n185' href='#n185'>185</a>
<a id='n186' href='#n186'>186</a>
<a id='n187' href='#n187'>187</a>
<a id='n188' href='#n188'>188</a>
<a id='n189' href='#n189'>189</a>
<a id='n190' href='#n190'>190</a>
<a id='n191' href='#n191'>191</a>
<a id='n192' href='#n192'>192</a>
<a id='n193' href='#n193'>193</a>
<a id='n194' href='#n194'>194</a>
<a id='n195' href='#n195'>195</a>
<a id='n196' href='#n196'>196</a>
<a id='n197' href='#n197'>197</a>
<a id='n198' href='#n198'>198</a>
<a id='n199' href='#n199'>199</a>
<a id='n200' href='#n200'>200</a>
<a id='n201' href='#n201'>201</a>
<a id='n202' href='#n202'>202</a>
<a id='n203' href='#n203'>203</a>
<a id='n204' href='#n204'>204</a>
<a id='n205' href='#n205'>205</a>
<a id='n206' href='#n206'>206</a>
<a id='n207' href='#n207'>207</a>
<a id='n208' href='#n208'>208</a>
<a id='n209' href='#n209'>209</a>
<a id='n210' href='#n210'>210</a>
<a id='n211' href='#n211'>211</a>
<a id='n212' href='#n212'>212</a>
<a id='n213' href='#n213'>213</a>
<a id='n214' href='#n214'>214</a>
<a id='n215' href='#n215'>215</a>
<a id='n216' href='#n216'>216</a>
<a id='n217' href='#n217'>217</a>
<a id='n218' href='#n218'>218</a>
<a id='n219' href='#n219'>219</a>
<a id='n220' href='#n220'>220</a>
<a id='n221' href='#n221'>221</a>
<a id='n222' href='#n222'>222</a>
<a id='n223' href='#n223'>223</a>
<a id='n224' href='#n224'>224</a>
<a id='n225' href='#n225'>225</a>
<a id='n226' href='#n226'>226</a>
<a id='n227' href='#n227'>227</a>
<a id='n228' href='#n228'>228</a>
<a id='n229' href='#n229'>229</a>
<a id='n230' href='#n230'>230</a>
<a id='n231' href='#n231'>231</a>
<a id='n232' href='#n232'>232</a>
<a id='n233' href='#n233'>233</a>
<a id='n234' href='#n234'>234</a>
<a id='n235' href='#n235'>235</a>
<a id='n236' href='#n236'>236</a>
<a id='n237' href='#n237'>237</a>
</pre></td>
<td class='lines'><pre><code><span style="color:#000000; font-weight:bold">--- ./third_party/lss/linux_syscall_support.h.orig</span>
<span style="color:#0057ae">+++ ./third_party/lss/linux_syscall_support.h</span>
<span style="color:#010181">&#64;&#64; -139,11 +139,13 &#64;&#64;</span>
  */
 #pragma push_macro(&quot;stat64&quot;)
 #pragma push_macro(&quot;fstat64&quot;)
<span style="color:#0057ae">+#pragma push_macro(&quot;fstatat64&quot;)</span>
 #pragma push_macro(&quot;lstat64&quot;)
 #pragma push_macro(&quot;pread64&quot;)
 #pragma push_macro(&quot;pwrite64&quot;)
 #undef stat64
 #undef fstat64
<span style="color:#0057ae">+#undef fstatat64</span>
 #undef lstat64
 #undef pread64
 #undef pwrite64
<span style="color:#010181">&#64;&#64; -1244,6 +1246,12 &#64;&#64;</span>
 #ifndef __NR_fallocate
 #define __NR_fallocate          285
 #endif
<span style="color:#0057ae">+</span>
<span style="color:#0057ae">+#undef __NR_pread</span>
<span style="color:#0057ae">+#define __NR_pread __NR_pread64</span>
<span style="color:#0057ae">+#undef __NR_pwrite</span>
<span style="color:#0057ae">+#define __NR_pwrite __NR_pwrite64</span>
<span style="color:#0057ae">+</span>
 /* End of x86-64 definitions                                                 */
 #elif defined(__mips__)
 #if _MIPS_SIM == _MIPS_SIM_ABI32
<span style="color:#010181">&#64;&#64; -4520,6 +4528,7 &#64;&#64;</span>
  * corresponding #pragma push_macro near the top of this file. */
 #pragma pop_macro(&quot;stat64&quot;)
 #pragma pop_macro(&quot;fstat64&quot;)
<span style="color:#0057ae">+#pragma pop_macro(&quot;fstatat64&quot;)</span>
 #pragma pop_macro(&quot;lstat64&quot;)
 #pragma pop_macro(&quot;pread64&quot;)
 #pragma pop_macro(&quot;pwrite64&quot;)
<span style="color:#000000; font-weight:bold">--- ./third_party/breakpad/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h.orig</span>
<span style="color:#0057ae">+++ ./third_party/breakpad/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h</span>
<span style="color:#010181">&#64;&#64; -37,6 +37,10 &#64;&#64;</span>
 #include &quot;common/memory.h&quot;
 #include &quot;google_breakpad/common/minidump_format.h&quot;
 
<span style="color:#0057ae">+#if !defined(__GLIBC__)</span>
<span style="color:#0057ae">+	#define _libc_fpstate _fpstate</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 namespace google_breakpad {
 
 // Wraps platform-dependent implementations of accessors to ucontext_t structs.
<span style="color:#000000; font-weight:bold">--- ./third_party/breakpad/breakpad/src/client/linux/minidump_writer/minidump_writer.h.orig</span>
<span style="color:#0057ae">+++ ./third_party/breakpad/breakpad/src/client/linux/minidump_writer/minidump_writer.h</span>
<span style="color:#010181">&#64;&#64; -48,7 +48,7 &#64;&#64;</span>
 #if defined(__aarch64__)
 typedef struct fpsimd_context fpstate_t;
 #elif !defined(__ARM_EABI__) &amp;&amp; !defined(__mips__)
<span style="color:#000000; font-weight:bold">-typedef struct _libc_fpstate fpstate_t;</span>
<span style="color:#0057ae">+typedef struct _fpstate fpstate_t;</span>
 #endif
 
 // These entries store a list of memory regions that the client wants included
<span style="color:#000000; font-weight:bold">--- ./third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h.orig</span>
<span style="color:#0057ae">+++ ./third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h</span>
<span style="color:#010181">&#64;&#64; -36,6 +36,7 &#64;&#64;</span>
 #include &lt;elf.h&gt;
 #include &lt;link.h&gt;
 #include &lt;stddef.h&gt;
<span style="color:#0057ae">+#include &lt;limits.h&gt;</span>
 
 #include &quot;common/memory_range.h&quot;
 
<span style="color:#010181">&#64;&#64; -51,9 +52,9 &#64;&#64;</span>
   typedef ElfW(Phdr) Phdr;
   typedef ElfW(Word) Word;
   typedef ElfW(Addr) Addr;
<span style="color:#000000; font-weight:bold">-#if __WORDSIZE == 32</span>
<span style="color:#0057ae">+#if ULONG_MAX == 0xffffffff</span>
   static const int kClass = ELFCLASS32;
<span style="color:#000000; font-weight:bold">-#elif __WORDSIZE == 64</span>
<span style="color:#0057ae">+#elif ULONG_MAX == 0xffffffffffffffff</span>
   static const int kClass = ELFCLASS64;
 #else
 #error &quot;Unsupported __WORDSIZE for ElfCoreDump.&quot;
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/suid/process_util.h.orig</span>
<span style="color:#0057ae">+++ ./sandbox/linux/suid/process_util.h</span>
<span style="color:#010181">&#64;&#64; -11,6 +11,14 &#64;&#64;</span>
 #include &lt;stdbool.h&gt;
 #include &lt;sys/types.h&gt;
 
<span style="color:#0057ae">+// Some additional functions</span>
<span style="color:#0057ae">+# define TEMP_FAILURE_RETRY(expression) \</span>
<span style="color:#0057ae">+	(__extension__			\</span>
<span style="color:#0057ae">+	 ({ long int __result;		\</span>
<span style="color:#0057ae">+	  do __result = (long int) (expression); \</span>
<span style="color:#0057ae">+	  while (__result == -1L &amp;&amp; errno == EINTR); \</span>
<span style="color:#0057ae">+	  __result; }))</span>
<span style="color:#0057ae">+</span>
 // This adjusts /proc/process/oom_score_adj so the Linux OOM killer
 // will prefer certain process types over others. The range for the
 // adjustment is [-1000, 1000], with [0, 1000] being user accessible.
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/seccomp-bpf/trap.cc.orig	2020-04-12 08:26:40.184159217 -0400</span>
<span style="color:#0057ae">+++ ./sandbox/linux/seccomp-bpf/trap.cc	2020-04-12 08:46:16.737191222 -0400</span>
<span style="color:#010181">&#64;&#64; -174,7 +174,7 &#64;&#64;</span>
   // If the version of glibc doesn&#39;t include this information in
   // siginfo_t (older than 2.17), we need to explicitly copy it
   // into an arch_sigsys structure.
<span style="color:#000000; font-weight:bold">-  memcpy(&amp;sigsys, &amp;info-&gt;_sifields, sizeof(sigsys));</span>
<span style="color:#0057ae">+  memcpy(&amp;sigsys, &amp;info-&gt;__sifields, sizeof(sigsys));</span>
 #endif
 
 #if defined(__mips__)
<span style="color:#000000; font-weight:bold">--- ./third_party/ffmpeg/libavutil/cpu.c.orig</span>
<span style="color:#0057ae">+++ ./third_party/ffmpeg/libavutil/cpu.c</span>
<span style="color:#010181">&#64;&#64; -38,7 +38,6 &#64;&#64;</span>
 #include &lt;sys/param.h&gt;
 #endif
 #include &lt;sys/types.h&gt;
<span style="color:#000000; font-weight:bold">-#include &lt;sys/sysctl.h&gt;</span>
 #endif
 #if HAVE_UNISTD_H
 #include &lt;unistd.h&gt;
diff --git a/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc b/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
<span style="color:#000000; font-weight:bold">--- chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc</span>
<span style="color:#0057ae">+++ chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc</span>
<span style="color:#010181">&#64;&#64; -39,7 +39,9 &#64;&#64;</span>
 #endif  // defined(OS_ANDROID) &amp;&amp; defined(__arm__)
 
 #if defined(OS_LINUX) &amp;&amp; !defined(OS_CHROMEOS)
<span style="color:#0057ae">+#if defined(__GLIBC__)</span>
 #include &lt;gnu/libc-version.h&gt;
<span style="color:#0057ae">+#endif</span>
 
 #include &quot;base/linux_util.h&quot;
 #include &quot;base/strings/string_split.h&quot;
&#64;&#64; -295,7 +297,7 &#64;&#64; void RecordLinuxDistro() {
 #endif  // defined(OS_LINUX) &amp;&amp; !defined(OS_CHROMEOS)
 
 void RecordLinuxGlibcVersion() {
<span style="color:#000000; font-weight:bold">-#if defined(OS_LINUX) &amp;&amp; !defined(OS_CHROMEOS)</span>
<span style="color:#0057ae">+#if defined(__GLIBC__) &amp;&amp; !defined(OS_CHROMEOS)</span>
   base::Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
<span style="color:#000000; font-weight:bold">--- services/device/serial/serial_io_handler_posix.cc.orig	2019-07-03 10:57:32.568171835 -0400</span>
<span style="color:#0057ae">+++ services/device/serial/serial_io_handler_posix.cc	2019-07-03 10:57:16.867983031 -0400</span>
<span style="color:#010181">&#64;&#64; -6,6 +6,7 &#64;&#64;</span>
 
 #include &lt;sys/ioctl.h&gt;
 #include &lt;termios.h&gt;
<span style="color:#0057ae">+#include &lt;asm-generic/ioctls.h&gt;</span>
 
 #include &lt;algorithm&gt;
 #include &lt;utility&gt;
diff --git a/third_party/ots/include/opentype-sanitiser.h b/third_party/ots/include/opentype-sanitiser.h
<span style="color:#000000; font-weight:bold">--- third_party/ots/include/opentype-sanitiser.h</span>
<span style="color:#0057ae">+++ third_party/ots/include/opentype-sanitiser.h</span>
&#64;&#64; -20,6 +20,7 &#64;&#64; typedef unsigned __int64 uint64_t;
 #define htonl(x) _byteswap_ulong (x)
 #define htons(x) _byteswap_ushort (x)
 #else
<span style="color:#0057ae">+#include &lt;sys/types.h&gt;</span>
 #include &lt;arpa/inet.h&gt;
 #include &lt;stdint.h&gt;
 #endif
<span style="color:#000000; font-weight:bold">--- ./base/logging.cc.orig</span>
<span style="color:#0057ae">+++ ./base/logging.cc</span>
<span style="color:#010181">&#64;&#64; -545,8 +545,7 &#64;&#64;</span>
 
 LogMessage::~LogMessage() {
   size_t stack_start = stream_.tellp();
<span style="color:#000000; font-weight:bold">-#if !defined(OFFICIAL_BUILD) &amp;&amp; !defined(OS_NACL) &amp;&amp; !defined(__UCLIBC__) &amp;&amp; \</span>
<span style="color:#000000; font-weight:bold">-    !defined(OS_AIX)</span>
<span style="color:#0057ae">+#if !defined(OFFICIAL_BUILD) &amp;&amp; !defined(OS_NACL) &amp;&amp; defined(__GLIBC__)</span>
   if (severity_ == LOG_FATAL &amp;&amp; !base::debug::BeingDebugged()) {
     // Include a stack trace on a fatal, unless a debugger is attached.
     base::debug::StackTrace trace;
<span style="color:#000000; font-weight:bold">--- ./third_party/blink/renderer/platform/wtf/stack_util.cc.orig</span>
<span style="color:#0057ae">+++ ./third_party/blink/renderer/platform/wtf/stack_util.cc</span>
<span style="color:#010181">&#64;&#64; -28,7 +28,7 &#64;&#64;</span>
 // FIXME: On Mac OSX and Linux, this method cannot estimate stack size
 // correctly for the main thread.
 
<span style="color:#000000; font-weight:bold">-#elif defined(__GLIBC__) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \</span>
<span style="color:#0057ae">+#elif defined(OS_LINUX) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \</span>
     defined(OS_FUCHSIA)
   // pthread_getattr_np() can fail if the thread is not invoked by
   // pthread_create() (e.g., the main thread of webkit_unit_tests).
<span style="color:#010181">&#64;&#64; -96,7 +96,7 &#64;&#64;</span>
 }
 
 void* GetStackStart() {
<span style="color:#000000; font-weight:bold">-#if defined(__GLIBC__) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \</span>
<span style="color:#0057ae">+#if defined(OS_LINUX) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \</span>
     defined(OS_FUCHSIA)
   pthread_attr_t attr;
   int error;
<span style="color:#000000; font-weight:bold">--- ./net/dns/dns_config_service_posix.cc.orig</span>
<span style="color:#0057ae">+++ ./net/dns/dns_config_service_posix.cc</span>
<span style="color:#010181">&#64;&#64; -122,7 +122,7 &#64;&#64;</span>
   ConfigParsePosixResult result;
   config-&gt;unhandled_options = false;
 // TODO(fuchsia): Use res_ninit() when it&#39;s implemented on Fuchsia.
<span style="color:#000000; font-weight:bold">-#if defined(OS_OPENBSD) || defined(OS_FUCHSIA)</span>
<span style="color:#0057ae">+#if defined(OS_OPENBSD) || defined(OS_FUCHSIA) || defined(_GNU_SOURCE)</span>
   // Note: res_ninit in glibc always returns 0 and sets RES_INIT.
   // res_init behaves the same way.
   memset(&amp;_res, 0, sizeof(_res));
<span style="color:#000000; font-weight:bold">--- third_party/swiftshader/third_party/llvm-subzero/lib/Support/Unix/Signals.inc.orig	2019-06-18 11:51:17.000000000 -0400</span>
<span style="color:#0057ae">+++ third_party/swiftshader/third_party/llvm-subzero/lib/Support/Unix/Signals.inc	2019-07-03 12:32:50.938758186 -0400</span>
<span style="color:#010181">&#64;&#64; -25,7 +25,7 &#64;&#64;</span>
 #include &quot;llvm/Support/raw_ostream.h&quot;
 #include &lt;algorithm&gt;
 #include &lt;string&gt;
<span style="color:#000000; font-weight:bold">-#if HAVE_EXECINFO_H</span>
<span style="color:#0057ae">+#if HAVE_EXECINFO_H &amp;&amp; defined(__GLIBC__)</span>
 # include &lt;execinfo.h&gt;         // For backtrace().
 #endif
 #if HAVE_SIGNAL_H
<span style="color:#010181">&#64;&#64; -52,6 +52,7 &#64;&#64;</span>
 #include &lt;unwind.h&gt;
 #else
 #undef HAVE__UNWIND_BACKTRACE
<span style="color:#0057ae">+#undef HAVE_BACKTRACE</span>
 #endif
 #endif
 
<span style="color:#000000; font-weight:bold">--- third_party/nasm/nasmlib/realpath.c.orig	2019-07-03 12:23:05.021949895 -0400</span>
<span style="color:#0057ae">+++ third_party/nasm/nasmlib/realpath.c	2019-07-03 12:24:24.246862665 -0400</span>
<span style="color:#010181">&#64;&#64; -49,7 +49,7 &#64;&#64;</span>
 
 #include &quot;nasmlib.h&quot;
 
<span style="color:#000000; font-weight:bold">-#ifdef HAVE_CANONICALIZE_FILE_NAME</span>
<span style="color:#0057ae">+#if defined(__GLIBC__)</span>
 
 /*
  * GNU-specific, but avoids the realpath(..., NULL)
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated by <a href='https://git.zx2c4.com/cgit/about/'>cgit v1.2.1</a> (<a href='https://git-scm.com/'>git 2.18.0</a>) at 2020-06-06 19:56:04 +0000</div>
</div> <!-- id=cgit -->
</body>
</html>
